### ISSUE TITLE
`TreeList` component renders no cell content

### ISSUE THREAD

**Issue by danielturus** [telerik/kendo-react#1396]
## I'm submitting a...

* Bug report 

## Current behavior
TreeList render no cell content on random page actions (in this case opening simple dialog).

Component: TreeList
Version: 5.5.0

## Expected behavior
TreeList should keep the same cell contents if no prop changed (with the same input should result same output)

## Minimal reproduction of the problem with instructions

1. Open the provided example in StackBlitz https://stackblitz.com/edit/react-jm4qu7?file=Modal.js,CustomTreeList.js,data.js,index.js
2. Click on "Open simple modal" button
3. The TreeView content disappears from the view because of this line of code (I suppose)
```js
if (this.prevColumns !== columns) {
            this.extendedColumn = readColumns(columns, { prevId: 0, idPrefix: idPrefix });
            this.columnsMap = mapColumns(this.extendedColumn);
}
```
Not sure here why two arrays are compared with equal, maybe that's the problem (`this.prevColumns` is of type Array, also `column`)

## Environment

Browser:
- Chrome (desktop) version: latest version 104

System:
- TypeScript version: N/A
- Node version: N/A
- Platform:  macOS

**Comment by danielturus** [comment]
From my findings, it seems that the component unmounts and then remounts (and because it unmounts, `this.extendedColumn = []` runs, which causes all the issues).

You can see below some info:
First render (did nothing on the UI):
<img width="767" alt="image" src="https://user-images.githubusercontent.com/32390499/186156972-f315fad5-da25-42b9-8725-fe2706d5fc14.png">

Second render (after clicking an icon to expand a node):
<img width="763" alt="image" src="https://user-images.githubusercontent.com/32390499/186157498-9f0daa42-d6aa-437b-938c-9734990237f5.png">

On second rerender, `this.prevColumns` and `this.extendedColumn` have unexpected values.

**Comment by nstoychev** [comment]
From my test, it will work if you remove the strict mode.

I'll check this later.

**Comment by danielturus** [comment]
Indeed, thank you! A fix with strict mode enabled would be great though!

**Comment by nstoychev** [comment]
Adding `this.prevColumns = [];` into `componentWillUnmount` component lifecycle function resolves the issue. I'll make a pull request with this fix.

**Comment by nstoychev** [comment]
Fixed in version 5.8.0.

Example: https://stackblitz.com/edit/react-jm4qu7-zrpkfy?file=index.js
